apps/projects/routes.py

```diff
@@ -1093,17 +1198,32 @@ def _get_timeline_thumbnails(self, amount):
             )
             return json_response({"processing": True}, status=202)
 
-    def _get_preview_thumbnail(self, position):
+    def _get_preview_thumbnail(self, position, crop, rotate):
         """
         Get or create thumbnail for preview
         :param position: video position to capture a frame
         :type position: int
+        :param crop: crop editing rules
+        :type crop: dict
+        :param rotate: rotate degree
+        :type rotate: int
         :return: json response
         :rtype: flask.wrappers.Response
         """
+        # validate crop param
```

Comment:
Let's cover this lines (1215-1222) with tests to keep 100% coverage for `routes.py`
You can get coverage report to see missing lines by running `pytest --cov-report term-missing --cov`